core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -413,6 +418,7 @@ private[spark] class ExecutorAllocationManager(
     private val stageIdToNumTasks = new mutable.HashMap[Int, Int]
     private val stageIdToTaskIndices = new mutable.HashMap[Int, mutable.HashSet[Int]]
     private val executorIdToTaskIds = new mutable.HashMap[String, mutable.HashSet[Long]]
+    private var numRunningTasks: Int = _
```

Comment:
what are the semantics here? Does this count speculation? In steady state is this guaranteed to go back to 0?